### DISCUSSION TITLE
Make RPi RAM into ProDOS ramdisk

### DISCUSSION THREAD

**Discussion by ajack2001my** [tjboldt/Apple2-IO-RPi#79] [General]
Can this project make the RPi's RAM into a RAMDISK via ProDOS akin to the slot 3 /RAM?

**Comment by tjboldt** [comment]
There wouldn’t be a significant speed difference from the current persisted virtual disk because there is still the overhead of transferring data between the Apple II data bus and the Raspberry Pi GPIO ports, but yes, it would be possible to host the virtual drive in RAM. I am actually working on support for more than two drives at a time and generating a virtual drive image in RAM based on the current working directory of the Linux host. That way you can access the Linux file system a bit more directly from ProDOS.

**Reply by ajack2001my** [reply]
I asked as the mechanism in ProDOS allows programs to disable the RAM disk as us it as additional memory. Much like how AppleWorks has so much RAM, or am I mistaken here? In any case, that's for the update. Will this also work on an Apple //gs or only 8-bit Apple //s?

**Comment by tjboldt** [comment]
The hardware design of the board does not permit the RPi RAM to be accessed directly so it won't help you with AppleWorks or any other program expecting a RAM card. This card works with a command and response pattern, so for example, ProDOS asks via the card's firmware to read or write a block. The firmware sends a command via the Apple II's data bus to latch chips and then to the GPIO ports of the RPi which are read by the driver running on the RPi. The driver responds and then the firmware copies to bytes to the buffer space ProDOS was expecting. The driver on the RPi also accepts other commands such running a shell with VT100 emulation in combination with the SHELL app on the default drive image. I have done some limited testing on an Apple IIgs with a TranswarpGS installed. It booted GS/OS successfully but that's about the extent of the testing I have done with it.
